Source/Extensions/Blazorise.DataGrid/DataGrid.razor.cs

```diff
@@ -1312,9 +1318,22 @@ private IEnumerable<TItem> FilterViewData()
                 CurrentPage = paginationContext.LastPage;
                 skipElements = ( CurrentPage - 1 ) * PageSize;
             }
-            return filteredData.Skip( skipElements ).Take( PageSize );
+
+            var pagedData = filteredData.Skip( skipElements ).Take( PageSize );
+
+            foreach ( var column in Columns )
+            {
+                if ( column.Groupable )
+                {
+                    //TODO : Need to think about this? Can we really have multiple GroupBy Funcs? Does that make sense at all? There should only be a single Func?

```

Comment:
We can have nested grouping. This is why grouping is on a column level.

eg

![image](https://user-images.githubusercontent.com/900302/210734473-c91bbe04-625b-4133-8918-a58db794d642.png)

Reply:
You re right I did not think right about the problem yesterday. See my comment regarding possible ordering to define the order the grouping / nesting happens?